src/lib/annotations/AnnotationThread.js

```diff
@@ -397,6 +378,38 @@ class AnnotationThread extends EventEmitter {
     //--------------------------------------------------------------------------
 
     /**
+     * Update a temporary annotation with the annotation saved on the backend and set the threadNumber if it does
+     * not exist. Propogate the threadnumber to dialog if applicable.
+     *
+     * @private
+     * @param {Annotation} serverAnnotation - The annotation returned by the backend. Use this as the source of truth
+     * @param {Annotation} localAnnotation - The temporary annotation holding place of the server validated annotation
```

Comment:
The temporary annotation stored locally holding place of the server validated annotation